src/algorithms.rs

```diff
@@ -1,4 +1,5 @@
 use std::str::FromStr;
+use self::Algorithm::*;
```

Comment:
I guess this hits too close to `using namespace <X>;` where `X` is at least three unrelated ones that clash in horrible ones.